src/csharp/Microsoft.Spark.E2ETest/IpcTests/Sql/DataFrameTests.cs

```diff
@@ -51,60 +48,100 @@ public void TestCollect()
-                        .Collect()
-                        .ToArray();
-                    Assert.Equal(3, rows.Length);
-                    Assert.Equal("Michael is 0", rows[0].GetAs<string>(0));
-                    Assert.Equal("hello Michael!", rows[0].GetAs<string>(1));
-
-                    Assert.Equal("Andy is 30", rows[1].GetAs<string>(0));
-                    Assert.Equal("hello Andy!", rows[1].GetAs<string>(1));
-
-                    Assert.Equal("Justin is 19", rows[2].GetAs<string>(0));
-                    Assert.Equal("hello Justin!", rows[2].GetAs<string>(1));
-                }
+        [Fact]
+        public void TestUDF()
+        {
+            // Single UDF.
+            Func<Column, Column, Column> udf1 = Udf<int?, string, string>(
+                (age, name) => name + " is " + (age ?? 0));
+            {
+                Row[] rows = _df.Select(udf1(_df["age"], _df["name"])).Collect().ToArray();
+                Assert.Equal(3, rows.Length);
+                Assert.Equal("Michael is 0", rows[0].GetAs<string>(0));
+                Assert.Equal("Andy is 30", rows[1].GetAs<string>(0));
+                Assert.Equal("Justin is 19", rows[2].GetAs<string>(0));
+            }
+
+            // Chained UDFs.
+            Func<Column, Column> udf2 = Udf<string, string>(str => $"hello {str}!");
+            {
+                Row[] rows = _df
+                    .Select(udf2(udf1(_df["age"], _df["name"])))
+                    .Collect()
+                    .ToArray();
+                Assert.Equal(3, rows.Length);
+                Assert.Equal("hello Michael is 0!", rows[0].GetAs<string>(0));
+                Assert.Equal("hello Andy is 30!", rows[1].GetAs<string>(0));
+                Assert.Equal("hello Justin is 19!", rows[2].GetAs<string>(0));
+            }
+
+            // Multiple UDFs:
+            {
+                Row[] rows = _df
+                    .Select(udf1(_df["age"], _df["name"]), udf2(_df["name"]))
+                    .Collect()
+                    .ToArray();
+                Assert.Equal(3, rows.Length);
+                Assert.Equal("Michael is 0", rows[0].GetAs<string>(0));
+                Assert.Equal("hello Michael!", rows[0].GetAs<string>(1));
+
+                Assert.Equal("Andy is 30", rows[1].GetAs<string>(0));
+                Assert.Equal("hello Andy!", rows[1].GetAs<string>(1));
+
+                Assert.Equal("Justin is 19", rows[2].GetAs<string>(0));
+                Assert.Equal("hello Justin!", rows[2].GetAs<string>(1));
             }
-            finally
-            {
-                SetUseArrowValue(originalUseArrow);
+        }
+
+        [Fact]
+        public void TestVectorUdf()
+        {
+            // Single UDF.
+            Func<Column, Column, Column> udf1 = VectorUdf<Int32Array, StringArray, StringArray>(
+                (ages, names) => (StringArray)ToArrowArray(
+                        Enumerable.Range(0, names.Length)

```

Comment:
alignment is off by 4. (Line 116 looks correct)

Reply:
Fixed. There were a few other places that were off as well.